doc/tutorials/basic/README.md

```diff
@@ -0,0 +1,303 @@
+
+# Creating Tests with JSonnet Templates
+
+## Prerequisites
+
+1. [A GCP project](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+1. A copy of this repository.
+1. Dependencies from [developing.md](/doc/developing.md).
+1. A GKE cluster with GPUs and (optionally) Cloud TPUs. Accelerator availability depends on GCE zone. All of the accelerators used in this tutorial are available in `us-central1-b`.
+  - Example commands:
+    ```bash
+    gcloud container clusters create tutorial-cluster \
+      --zone us-central1-b \
+      --release-channel regular \
+      --machine-type n1-standard-4 \
+      --accelerator "type=nvidia-tesla-v100,count=1" \
+      --scopes "https://www.googleapis.com/auth/cloud-platform" \
+      --num-nodes 1 \
+      --enable-ip-alias \
+      --enable-autoupgrade \
+      --enable-tpu \
+      --project=$PROJECT_ID
+    ```
+    ```bash
+    # Connect to your cluster
+    gcloud container clusters get-credentials tutorial-cluster --project $PROJECT_ID --zone us-central1-b
+    ```
+    ```bash
+    # Install GPU drivers
+    kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+    ```
+1. A GCS bucket.
+  - To create a new GCS bucket, run `gsutil mb -c standard -l us-central1 gs://your-bucket-name`
+
+`ml-testing-accelerators` relies heavily on GKE to run workloads, so it is also expected that you should have basic familiarity with creating and running Kubernetes batch workloads. If you need a refresher, see the following documents:
+
+- [Learn Kubernetes Basics](https://kubernetes.io/docs/tutorials/kubernetes-basics/)
+- [Pods](https://kubernetes.io/docs/concepts/workloads/pods/) (basic unit of a workload)
+- [Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job/) (batch workloads)
+
+`ml-testing-accelerators` also uses [JSonnet](https://jsonnet.org) templates to build Kubernetes workloads. It may be helpful to work through [JSonnet's tutorials](https://jsonnet.org/learning/tutorial.html) first, but the examples in this document are simple enough that it is not required.
+
+Before you begin, set the following environment variables:
+
+```bash
+$ export PROJECT_ID=...
+$ export GCS_BUCKET=gs://...
+```
+
+## Creating a simple test without accelerators
```

Comment:
This might just be personal feeling - I feel it is helpful that the doc can explain at a high level what it is doing. My experience with following this doc is first pass I was blindly copying/pasting the command, but only after checking pantheon and logs, I get a rough understanding on the workflow.

To summarize my own understanding, which could be wrong:
```
The workflow first builds a docker image containing your repo's latest code and dependencies, and then specifies command in jsonnet, which tells GKE cluster what commands to run (usually some tests in the repo). The running results can be viewed in the GKE running log.
```